open_ce/build_tree.py

```diff
@@ -108,93 +114,96 @@ def name(self):
         result = result.replace("_", "-")
         return result
 
-    def all_outputs_exist(self, output_folder):
-        """
-        Returns true if all of the output_files already exist.
-        """
-        return all((os.path.exists(os.path.join(os.path.abspath(output_folder), package))
-                    for package in self.output_files))
+    def __repr__(self):
+        return str(self.__key())
+
+    def __str__(self):
+        return self.name()
 
     def __key(self):
-        return (self.recipe, self.output_files)
+        return (self.name(), ",".join(self.output_files))
 
     def __hash__(self):
-        return hash(self.__key())
+        return hash(self.__repr__())
 
     def __eq__(self, other):
         if not isinstance(other, BuildCommand):
             return False
         return self.__key() == other.__key()  # pylint: disable=protected-access
 
-def traverse_build_commands(commands, deps):
+    def get_all_dependencies(self):
+        '''
+        Return a union of all dependencies.
+        '''
+        return self.run_dependencies.union(self.host_dependencies, self.build_dependencies, self.test_dependencies)
+
+class DependencyNode():
     """
-    Generator function that goes through a list of BuildCommand's dependency tree.
+    Contains information for the dependency tree.
     """
-    if not deps:
-        deps = range(len(commands))
-    is_seen = [False for i in range(len(commands))]
-    yield from _traverse_build_commands(commands, is_seen, deps)
-
-def _traverse_build_commands(commands, is_seen, deps):
-    for dep in deps:
-        if is_seen[dep]:
-            continue
-        yield from _traverse_build_commands(commands, is_seen, commands[dep].build_command_dependencies)
-        yield commands[dep]
-        is_seen[dep] = True
+    #pylint: disable=too-many-instance-attributes,too-many-arguments,too-many-locals
```

Comment:
I don't think all of these pylint exceptions are needed here.

Reply:
You're right. These got copy and pasted.